.github/workflows/ci.yml

```diff
@@ -42,29 +42,45 @@ jobs:
           wget -qO- get.nextflow.io | bash
           sudo mv nextflow /usr/local/bin/
 
-      - name: Run pipeline with test data
-        run: |
-          nextflow run ${GITHUB_WORKSPACE} -profile test,docker
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.x"
+      - name: Install dependencies
+        run: python -m pip install --upgrade pip pytest-workflow
+
+      - name: Run pipeline with default settings
+        run: pytest --tag default --kwdof
+
+      - name: Upload logs on failure
+        if: failure()
+        uses: actions/upload-artifact@v2
+        with:
+          name: logs-default-${{ matrix.nxf_version }}
+          path: |
+            /tmp/pytest_workflow_*/*/.nextflow.log
+            /tmp/pytest_workflow_*/*/log.out
+            /tmp/pytest_workflow_*/*/log.err
 
   star_salmon:
-    name: Test STAR Salmon with workflow parameters
+    name: Test STAR Salmon with ${{ matrix.parameters }}
```

Comment:
Do we need a separate `star_salmon` for this? Couldn't this just be another entry in the `matrix`?

Reply:
It could be another entry in the matrix, I was thinking this was a good interim to make sure we didn't miss anything in the conversion.